File: code/api/Hubly.Api/DTOs/Company/GetCompanyOutputModel.cs
namespace Hubly.api.DTOs;

public class GetCompanyOutputModel
{
    public int Id { get; set; }
    public string CompanyName { get; set; } = string.Empty;
    public bool IsVerified { get; set; }
    public string? Description { get; set; }
    public string? Sector { get; set; }
    public string? CompanySize { get; set; }
    public string? WebsiteLink { get; set; }
    public string? CountryHeadquarters { get; set; }
}

File: code/web/services/DTO/company/GetCompanyOutputModel.ts
export class GetCompanyOutputModel {
    id: number;
    companyName: string;
    isVerified: boolean;
    description: string;
    sectors: string[]; // Corrigido para sintaxe TypeScript
    companySize: string;
    websiteLink: string;
    countryHeadquarters: string;

    constructor(data: any) {
        this.id = data.id;
        this.companyName = data.companyName;
        this.isVerified = data.isVerified;
        this.description = data.description;
        this.sectors = data.sectors || []; // Garante que inicia como array vazio se vier nulo
        this.companySize = data.companySize;
        this.websiteLink = data.websiteLink;
        this.countryHeadquarters = data.countryHeadquarters;
    }
}

export default GetCompanyOutputModel;